File: plain/src/main/java/com/github/daneko/android/plain/WeatherApi.java
package com.github.daneko.android.plain;

import rx.Observable;

import retrofit.http.GET;
import retrofit.http.Query;

/**
 *
 */
public interface WeatherApi {

    static final String OPEN_WEATHER_MAP_API = "http://api.openweathermap.org";

    @GET("/data/2.5/weather")
    Observable<WeatherResponse> get(@Query("q") String q, @Query("appid") String apiKey);
}

File: kotlin/src/main/kotlin/com/github/daneko/android/kotlin/WeatherApi.kt
package com.github.daneko.android.kotlin

import retrofit.http.GET
import retrofit.http.Query
import rx.Observable

/**

 */
interface WeatherApi {

    @GET("/data/2.5/weather")
    fun get(@Query("q") q: String, @Query("appid") apiKey: String): Observable<WeatherResponse>

    companion object {
        val OPEN_WEATHER_MAP_API = "http://api.openweathermap.org"
    }
}
